src/compositor.rs

```diff
@@ -103,55 +152,104 @@ macro_rules! delegate_compositor {
             ] => $crate::compositor::CompositorState
         );
     };
+    ($ty: ty, surface: $surface: ty) => {
```

Comment:
This macro has limited usefullness if we have two user data implementations for something in a user's app.

Reply:
Hmm, it might be useful to change that to a list of types instead of a single type.